packages/chat-stateful-client/src/ResourceDownloadQueue.ts

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { ChatContext } from './ChatContext';
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { ChatError } from './ChatClientState';
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { ChatMessageWithStatus } from './types/ChatMessageWithStatus';
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+import { CommunicationTokenCredential } from '@azure/communication-common';
+
+/* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+/**
+ * @private
+ */
+export class ResourceDownloadQueue {
+  private _messageQueue: ChatMessageWithStatus[] = [];
+  private _context: ChatContext;
+  private _isRequesting = false;
+  private _credential: CommunicationTokenCredential;
```

Comment:
Will this always be this type, or do we need a more generic HTTP auth header retrieval function?

Reply:
For the time being since everything we request is going up to AMS it would always need to be this type.

Reply:
i can confirm that for file download, it would be acs communication token as well

Reply:
What happens if the token expires during operation?
How do we handle a case where the fetch operation requires a different format than the standard authorization header?

If we had a function call to retrieve the expected header string, we can ensure that the auth header can be kept current with wherever the auth token is updated.

Reply:
> What happens if the token expires during operation?

This case is true for any calls we make to SDK. But our token credential has the ability (if enabled by developers) to refresh itself proactively. 

> How do we handle a case where the fetch operation requires a different format than the standard authorization header?

I think its important to remember in this case, this is strictly internal, developers won't be allowed to change the headers since we don't expose an API anyways. If this is a concern we will need to rethink this implementation and provide an API to allow this.